File: coursework/coursework2/load.py
import ast,re
def load_points(filename):
    with open(filename, 'r') as file:
        points=[]
        for line in file:
            label, point_str = line.strip().split(maxsplit=1)
            point_str = re.sub(r'\s+', ',', point_str)
            point = ast.literal_eval(point_str)
            points.append([label,point])
    return points
points=load_points("points.txt")
for point in points:
    print(point)

File: coursework/coursework2/load.jl
function load_points(filename)
    points_dict = Dict{NTuple{5, Float64}, String}()   
    open(filename, "r") do file
        for line in eachline(file)
            parts = split(replace(line, ['[', ']'] => ""))
            label = parts[1]
            point_values = [parse(Float64,parts[i]) for i in 2:6]
            point=tuple(point_values...)
            points_dict[point] = label
        end
    end
    return points_dict
end

# Test the function
points_dict = load_points("points.txt")

# Display the loaded points and labels
for (point, label) in points_dict
    println("$point -> $label")
end
